csrc/kernel_cache.cpp

```diff
@@ -201,7 +222,55 @@ std::vector<at::Tensor> FusionExecutorCache::runFusionWithInputs(
 
   KernelArgumentHolder args = prepareInputs(perm_inputs);
 
-  auto kernel_runtime = getKernelRuntimeFor(args, forced_index_type);
+  // Concretizing the fusion using args
+  auto fusion = fusion_.get();
+  if (has_dynamic_reshape_) {
+    ExpressionEvaluator expr_eval;
```

Comment:
I think `SchedulerRuntimeInfo` also has an expr evaluator. Can it be shared?

Reply:
Oh good idea. We'd have to figure out how to use it before we have the runtime kernel since that depends on which concrete kernel we have.